Automattic/jetpack
# Backup: Add next daily backup schedule time

Resolves https://github.com/Automattic/jetpack-backup-team/issues/701

## Proposed changes:

* Setup react-query, moment and typescript packages on the backup package
* Add `/jetpack/v4/site/backup/schedule` endpoint that will fetch the daily schedule backup time for a site
* Add `useScheduledTimeQuery` hook to fetch schedule time
* Add `useNextBackupSchedule` hook to parse and return the next backup schedule time
* Add `NextScheduledBackup` component to render the next scheduled backup message and render it on Backups main component

https://github.com/user-attachments/assets/38add82e-348a-47ac-94b3-f721fd32c508

### Other information:


## Jetpack product discussion


peaFOp-2Lz-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Spin up a Jurassic Ninja site using this branch: https://jurassic.ninja/create?jetpack-beta&branches.jetpack-backup=add/jetpack-backup-schedule-time&wp-debug-log&nojetpack
* Navigate to the VaultPress Backup page.
* If you have a successful backup recently, you should see when the next scheduled backup will run under the latest backup section, similar to this:
  ![CleanShot 2024-10-25 at 22 12 10@2x](https://github.com/user-attachments/assets/336f5fc9-e150-4bfc-a86a-8f9aff1a624a)
* Ensure that clicking on `Modify` record the `jetpack_backup_schedule_modify_click` Tracks event and it opens a new tab navigating to Jetpack Cloud settings page.
